Tests/SuperLinq.Test/FallbackIfEmptyTest.cs

```diff
@@ -10,6 +10,14 @@ public void FallbackIfEmptyWithEmptySequence()
 		source.FallbackIfEmpty(12, 23).AssertSequenceEqual(12, 23);
 	}
 
+	[Fact]
+	public void FallbackIfEmptyWithEmptyListSequence()
+	{
+		using var source = Enumerable.Empty<int>().ToList().AsTestingSequence(maxEnumerations: 2);
```

Comment:
not working, because TestingSequence is not recognised by the TryGetCollectionCount, so my test does not go through the new code

Reply:
i removed the testing sequence, otherwise the test was not taking the right path.

Might be a good idea to add a CollectionTestingSequence that implements ICollection for all those cases where we use TryGetCollectionCount